src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.QueueItem.cs

```diff
@@ -5,59 +5,95 @@
 #nullable enable
 
 using System;
-using System.Collections.Immutable;
 using System.Diagnostics;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Threading;
 
 namespace Microsoft.CodeAnalysis.LanguageServer.Handler
 {
     internal partial class RequestExecutionQueue
     {
-        private readonly struct QueueItem
+        private interface QueueItem
         {
             /// <summary>
-            /// Callback to call into underlying <see cref="IRequestHandler"/> to perform the actual work of this item.
+            /// Begins executing the work specified by this queue item.
             /// </summary>
-            private readonly Func<RequestContext?, CancellationToken, Task> _callbackAsync;
+            public Task CallbackAsync(RequestContext? context, CancellationToken cancellationToken);
```

Comment:
All of these access modifiers should probably be removed

```suggestion
            Task CallbackAsync(RequestContext? context, CancellationToken cancellationToken);
```

Reply:
remind me.  when do we call the callback when RequestContext is null? (feel free to doc this as well).